sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -896,6 +896,19 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
     checkError(df("`a.b`.c.`d"))
   }
 
+  test("SPARK-11977: Support accessing a column contains '.' without backticks") {
+    val df = sparkContext.parallelize(
+      (1, 2) :: (3, 4) ::  Nil).toDF("test.column1", "test.column.2")
+    checkAnswer(df.select("test.column1"), Seq(Row(1), Row(3)))
```

Comment:
This is a behaviour change, users may expect to see an analyse failure here. cc @rxin Do we want this feature?

Personally I'd like to have a new function(similar to `col`, but resolves the column by the given name directly) instead of changing the semantic of `select`